### ISSUE TITLE
Retry only for  specifc queries like select (get)

### ISSUE THREAD

**Issue by hadasafelman** [sequelize/sequelize#13931]
## Issue Creation Checklist

I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Feature Description

add to retry object
the option to do retry for only specific kind of quires

### Is your feature request related to a problem? Please describe.

when I run transaction with insert query -
the sql query run well, but we get reconnect error.
then the retry management will try to run the transaction again 
and get error since we cannot run this insert with same id twice

### Describe the solution you'd like

add to retry object queryTypes:[select,insert,create]

```js


const sequelize = new Sequelize(process.env.REMOTE_DB_NAME, 
    process.env.REMOTE_DB_NAME, 
    process.env.REMOTE_DB_PW, 
    {
    host: process.env.REMOTE_DB_HOST,
    dialect: 'mysql',
    // Here:
    retry: {
        match: [Sequelize.ConnectionError],
        queryTypes:[select,insert,create]

    }
})
```

### Why should this be in Sequelize

cannot use this retry management if I need to wrap it in different module

### Is this feature dialect-specific?

- [ ] No. This feature is relevant to Sequelize as a whole.

### Would you be willing to implement this feature by submitting a Pull Request?

- [ ] No, I don't have the time and I wouldn't even know how to start.

**Comment by ephys** [comment]
You can already control retry per-query but there isn't a way currently to do it per queryType.
You could disable retry globally then only enable it on relevant queries.

I think we could change the interface of `RetryOptions` to something that accepts more filtering parameters:

```typescript
// RetryMatcher is a function that determines whether a given query should be re-tried based on queryType & error.
type RetryMatcher = (params: { queryType: QueryType, error: Error }) => boolean;

export interface RetryOptions {
  match?: (RegExp | string | typeof Error)[] | RetryMatcher;
  max?: number;
}
```

PR welcome

**Comment by hadasafelman** [comment]
are u going to add this?
if so when u think u will be able to do it?

**Comment by ephys** [comment]
It's not on my personal backlog, but feel free to open a PR that implements it and we'll review & merge it

**Comment by WikiRik** [comment]
@ephys might be nice to reopen this issue?

**Comment by ephys** [comment]
Yup! I think it's a good task for newcomers if someone wants to give it a try
